lms/djangoapps/courseware/access.py

```diff
@@ -618,13 +618,18 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
+        sneakpeek_access = ACCESS_DENIED
         if user.is_authenticated():
             if not UserProfile.has_registered(user):
                 if _can_load_descriptor_nonregistered(descriptor):
-                    return ACCESS_GRANTED
+                    sneakpeek_access = ACCESS_GRANTED
         response = (
             _visible_to_nonstaff_users(descriptor)
-            and _has_group_access(descriptor, user, course_key)
+            and
+            (
+                _has_group_access(descriptor, user, course_key)
+                or sneakpeek_access
```

Comment:
> section 3 is hidden/staff only

I'm still seeing Section 3 in the screenshot.

I don't think this is correct. Remember, Direct Access users really are enrolled users, with an entry in `auth_user`.
What this expression is saying is:

> If their group deserves access, let them in.
> Even if their group doesn't have access, go ahead and let them in anyway, if they're sneak_peek.

So I suspect that sneak_peek users will mistakenly slide through if they're in a "normal group" that doesn't have access.

I don't think `_has_group_access` and `sneakpeek_access` are logically connected, so I'm not sure that grouping them together with an `or` is appropriate.

If you look at `_has_group_access` [1], any check that can `return ACCESS_DENIED` will be ignored for sneak_peek users. This one in particular jumps out [2]:
```python
log.warning("Group access check excludes all students, access will be denied.", exc_info=True)
```

[1] https://github.com/Stanford-Online/edx-platform/blob/giulio/sneakpeek-release-date-bugfix/lms/djangoapps/courseware/access.py#L537
[2] https://github.com/Stanford-Online/edx-platform/blob/giulio/sneakpeek-release-date-bugfix/lms/djangoapps/courseware/access.py#L554